split_settings/tools.py

```diff
@@ -108,10 +112,23 @@ def include(  # noqa: WPS210, WPS231, C901
 
             scope[_INCLUDED_FILE] = included_file
             with open(included_file, 'rb') as to_compile:
-                compiled_code = compile(  # noqa: WPS421
-                    to_compile.read(), included_file, 'exec',
-                )
-                exec(compiled_code, scope)  # noqa: S102, WPS421
+                if included_file.endswith('.py'):
```

Comment:
My proposal is different. I propose adding `compiled()` class as we do with `optional`, this way people would have to use it if they want to include a compiled file.

And there we can validate that path ends with `.pyc`

Reply:
I've defined a helper function `compiled` and class `_Compiled` (similar to `optional`/`_Optional` in 4fff98f.